moto/panorama/models.py

```diff
@@ -0,0 +1,304 @@
+import base64
+import json
+from datetime import datetime, timedelta
+from dateutil.tz import tzutc
+from typing import Dict, Any, Optional, Union, List
+from typing_extensions import Literal
+
+from moto.core import BackendDict, BaseBackend, BaseModel
+from moto.moto_api._internal.managed_state_model import ManagedState
+from moto.panorama.type import DEVICE_AGGREGATED_STATUS_TYPE, PROVISIONING_STATUS_TYPE
+from moto.panorama.utils import deep_convert_datetime_to_isoformat, hash_device_name
+from moto.utilities.paginator import paginate
+
+from .exceptions import (
+    ValidationError,
+)
+
+PAGINATION_MODEL = {
+    "list_devices": {
+        "input_token": "next_token",
+        "limit_key": "max_results",
+        "limit_default": 123,
+        "unique_attribute": "device_id",
+    },
+}
+
+
+class BaseObject(BaseModel):
+    def camelCase(self, key: str) -> str:
+        words = []
+        for word in key.split("_"):
+            words.append(word.title())
+        return "".join(words)
+
+    def update(self, details_json: str) -> None:
+        details = json.loads(details_json)
+        for k in details.keys():
+            setattr(self, k, details[k])
+
+    def gen_response_object(self) -> Dict[str, Any]:
+        response_object: Dict[str, Any] = dict()
+        for key, value in self.__dict__.items():
+            if "_" in key:
+                response_object[self.camelCase(key)] = value
+            else:
+                response_object[key[0].upper() + key[1:]] = value
+        return response_object
+
+    def response_object(self) -> Dict[str, Any]:  # type: ignore[misc]
+        return self.gen_response_object()
+
+
+class Device(BaseObject):
+    def __init__(
+        self,
+        account_id: str,
+        region_name: str,
+        description: Optional[str],
+        name: str,
+        network_configuration: Optional[Dict[str, Any]],
+        tags: Optional[Dict[str, str]],
+    ) -> None:
+        self.__device_aggregated_status_manager = ManagedState(
```

Comment:
Normally the Device-class would extend `ManagedState`, giving us the functionality out of the box. 

It took me a while to understand the reasoning behind your approach, but I now understand that's the only way we can keep track of two statuses. 

Could you leave some comments to explain this? That would make it easier to understand next time a developer is looking at this.

Reply:
Yes it is a good idea: https://github.com/getmoto/moto/pull/6948/commits/ff562d22b7a6f5bdfba5bc7b5e1dc074dfd50934

Tell me if you think that I should explain how to set them, make them advance etc